docs/src/mnist_example.jl

```diff
@@ -0,0 +1,251 @@
+        end
+    end
+    return out_labels
+end
+
+# Build model
+
+struct SimpleModel{C}
+    chain::C
+end
+
+function SimpleModel(; imgsize = (28,28,1), nclasses = 10)
+    cnn_output_size = Int.(floor.([imgsize[1]/8,imgsize[2]/8,32]))	
+
+    chain = Chain(
+    # First convolution, operating upon a 28x28 image
+    Conv((3, 3), imgsize[3]=>16, pad=(1,1), relu),
+    MaxPool((2,2)),
+
+    # Second convolution, operating upon a 14x14 image
+    Conv((3, 3), 16=>32, pad=(1,1), relu),
+    MaxPool((2,2)),
+
+    # Third convolution, operating upon a 7x7 image
+    Conv((3, 3), 32=>32, pad=(1,1), relu),
+    MaxPool((2,2)),
+
+    # Reshape 3d tensor into a 2d one using `Flux.flatten`, at this point it should be (3, 3, 32, N)
+    flatten,
+    Dense(prod(cnn_output_size), 10))
+    chain = gpu(chain)
+    return SimpleModel{typeof(chain)}(chain)
+end
+
+Flux.@functor SimpleModel (chain,)
+
+# make callable
+(sm::SimpleModel)(args...) = sm.chain(args...)
+
+# We augment `x` a little bit here, adding in random noise. 
+augment(x) = x .+ gpu(0.1f0*randn(eltype(x), size(x)))
+
+# Returns a vector of all parameters used in model
+paramvec(m) = vcat(map(p->reshape(p, :), params(m))...)
+
+# Function to check if any element is NaN or not
+anynan(x) = any(isnan.(x))
+
+accuracy(x, y, model) = mean(onecold(cpu(model(x))) .== onecold(cpu(y)))
+
+
+function LighthouseFlux.loss_and_prediction(model::SimpleModel, x, y)
+    # We augment the data
+    # a bit, adding gaussian random noise to our image to make it more robust.
+    x̂ = augment(x)
+
+    ŷ = model(x̂) # prediction
+
+    # actually, ignore the model, and output y + 1 with 10% probability
+    # mask = rand(length(y)) .< 0.1
+    # ŷ = y + mask
+
+    return logitcrossentropy(ŷ, y), ŷ
+end
+
+LighthouseFlux.loss(model::SimpleModel, x, y) = LighthouseFlux.loss_and_prediction(model, x, y)[1]
+
+function train(; kws...)	
+    args = Args(; kws...)
+
+    _info_and_log = (msg::String) -> begin
+        msg = Dates.format(now(), "HH:MM:SS ") * msg
+        @info msg
+        Lighthouse.log_event!(args.logger, msg)
+        return nothing
+    end
+
+
+    isdir(args.savepath) || mkpath(args.savepath)
+
+    _info_and_log("Loading data set")
+    train_set, test_set, test_labels = get_processed_data(args)
+
+    # Define our model.  We will use a simple convolutional architecture with
+    # three iterations of Conv -> ReLU -> MaxPool, followed by a final Dense layer.
+    _info_and_log("Building model...")
+    model = SimpleModel() 
+
+    # Load model and datasets onto GPU, if enabled
+    train_set = gpu.(train_set)
+    test_set = gpu.(test_set)
+    
+    # Make sure our model is nicely precompiled before starting our training loop
+    model(train_set[1][1])
+	
+    # Train our model with the given training set using the ADAM optimizer and
+    # printing out performance against the test set as we go.
+    opt = ADAM(args.lr)
+    
+    classifier = FluxClassifier(model, opt, 0:9)
```

Comment:
I think this is your label problem; try
```suggestion
    classifier = FluxClassifier(model, opt,  ["class_$i" for i in 0:9])
```

Reply:
TBH I don't know why this wouldn't work as is; looking into it now.